test.js

```diff
@@ -0,0 +1,210 @@
+'use strict'
+
+const t = require('tap')
+const test = t.test
+const request = require('request')
+const Fastify = require('fastify')
+const fs = require('fs')
+
+test('reply.view exist', t => {
+  t.plan(6)
+  const fastify = Fastify()
+
+  fastify.register(require('./index'), {
+    engine: {
+      ejs: require('ejs')
+    }
+  })
+
+  fastify.get('/', (req, reply) => {
+    t.ok(reply.view)
+    reply.send({ hello: 'world' })
+  })
+
+  fastify.listen(0, err => {
+    t.error(err)
+
+    request({
+      method: 'GET',
+      uri: 'http://localhost:' + fastify.server.address().port
+    }, (err, response, body) => {
+      t.error(err)
+      t.strictEqual(response.statusCode, 200)
+      t.strictEqual(response.headers['content-length'], '' + body.length)
+      t.deepEqual(JSON.parse(body), { hello: 'world' })
+      fastify.close()
+    })
+  })
+})
+
+test('reply.view should return 500 if some data is missing', t => {
+  t.plan(3)
+  const fastify = Fastify()
+
+  fastify.register(require('./index'), {
+    engine: {
+      ejs: require('ejs')
+    }
+  })
+
+  fastify.get('/', (req, reply) => {
+    reply.view()
+  })
+
+  fastify.listen(0, err => {
+    t.error(err)
+
+    request({
+      method: 'GET',
+      uri: 'http://localhost:' + fastify.server.address().port
+    }, (err, response, body) => {
+      t.error(err)
+      t.strictEqual(response.statusCode, 500)
+      fastify.close()
+    })
+  })
+})
+
+/* test('register callback should throw if the engine is missing', t => {
+  t.plan(2)
+  const fastify = Fastify()
+
+  fastify.register(require('./index'), {}, err => {
+    t.error(err)
+    t.is(err.message, 'Missing engine')
+    t.pass()
+  })
+}) */
```

Comment:
I wanted to put your attention here, if I uncomment that test, it will throw en error and the test will fail.
```
test.js ............................................. 35/36
  not ok Missing engine
    stack: |
      fastifyView (index.js:13:10)
      Plugin.exec (node_modules/avvio/boot.js:191:3)
      Boot.loadPlugin (node_modules/avvio/boot.js:205:10)
      release (node_modules/fastq/queue.js:127:16)
      Object.resume (node_modules/fastq/queue.js:61:7)
    at:
      line: 13
      column: 10
      file: index.js
      function: fastifyView
    test: register callback should throw if the engine is missing
    source: |
      next(new Error('Missing engine'))

total ............................................... 35/36
```
I guess it happens because of [this](https://github.com/mcollina/avvio/blob/master/boot.js#L194-L198).